asv_bench/benchmarks/utils.py

```diff
@@ -248,14 +270,32 @@ def gen_data(
     -------
     dict
         Number of keys - `ncols`, each of them store np.ndarray of `nrows` length.
-        When `data_type`=="str_int" some of the columns will be of string type.
+
+    Notes
+    -----
+    Returned data type depends on the `data_type` parameter in the next way:
+    - `data_type`=="int" - all columns will be contain only integer values;
+    - `data_type`=="str_int" some of the columns will be of string type;
+    - `data_type`=="true_false_int" half of the columns will be filled with
+      string values representing "true" and "false" values and another half - with
+      integers.
     """
-    if data_type == "int":
-        return gen_int_data(nrows, ncols, rand_low, rand_high)
-    elif data_type == "str_int":
-        return gen_str_int_data(nrows, ncols, rand_low, rand_high)
-    else:
-        assert False
+    cache_key = (data_type, nrows, ncols, rand_low, rand_high)
+    if cache_key in data_cache:
+        return data_cache[cache_key]
+
+    type_to_generator = {
+        "int": gen_int_data,
+        "str_int": gen_str_int_data,
+        "true_false_int": gen_true_false_int_data,
+    }
+    data_generator = type_to_generator.get(data_type, None)
```

Comment:
`logging.info` also can be moved from `gen_int_data`, `gen_str_int_data` and `gen_true_false_int_data` to place between 291-292 lines (for example)

Reply:
Ok, implemented.